ReGathering_project/app/src/main/java/com/cmpt362/regathering/fragment/FragmentHome.kt

```diff
@@ -4,47 +4,98 @@ import android.os.Bundle
+class FragmentHome: Fragment(),
+    EventAdapter.OnEventSelectedListener {
+    companion object {
+        private const val LIMIT = 50
+    }
+
+    lateinit var firestore: FirebaseFirestore
+    lateinit var eventsQuery: Query
+
+    private lateinit var binding: FragmentHomeBinding
+    lateinit var eventAdapter: EventAdapter
+
+    private lateinit var fragmentView: View
+//    private lateinit var eventArrayAdapter: EventListAdapter
+    private lateinit var listViewEvents: ListView
+    private lateinit var eventArrayList: ArrayList<Event>
+
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        val view = inflater.inflate(R.layout.fragment_home, container, false)
-        listViewUpcomingEvents = view.findViewById(R.id.upcoming_events_list_view)
-        listViewSuggestedEvents = view.findViewById(R.id.suggested_events_list_view)
-        listUpcomingEvents = ArrayList()
-        listSuggestedEvents = ArrayList()
-        val arrayAdapterUpcoming = ArrayAdapter(requireActivity(), android.R.layout.simple_list_item_1, listUpcomingEvents)
-        listViewUpcomingEvents.adapter = arrayAdapterUpcoming
-        val arrayAdapterSuggested = ArrayAdapter(requireActivity(), android.R.layout.simple_list_item_1, listSuggestedEvents)
-        listViewSuggestedEvents.adapter = arrayAdapterSuggested
-        listUpcomingEvents.addAll(UPCOMING_LIST)
-        listSuggestedEvents.addAll(SUGGESTED_LIST)
-        return view
+    ): View {
+        binding = FragmentHomeBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // Get instance of firestore
+        firestore = Firebase.firestore
+
+        eventsQuery = firestore.collection("events")
+            .orderBy("date", Query.Direction.ASCENDING)
+            .limit(LIMIT.toLong())
+
+        // RecyclerView
+        eventAdapter = object : EventAdapter(eventsQuery, this@FragmentHome) {
+            override fun onDataChanged() {
+                // Show/hide content if the query returns empty.
+                if (itemCount == 0) {
+                    binding.recyclerEvents.visibility = View.GONE
+                } else {
+                    binding.recyclerEvents.visibility = View.VISIBLE
+                }
+            }
+
+            override fun onError(e: FirebaseFirestoreException) {
+                // Show a snackbar on errors
+                Snackbar.make(binding.root,
+                    "Error: check logs for info.", Snackbar.LENGTH_LONG).show()
+            }
+        }
+
+        val layoutManager = LinearLayoutManager(context)
+        binding.recyclerEvents.layoutManager = layoutManager
+        binding.recyclerEvents.adapter = eventAdapter
+    }
```

Comment:
Could you please create an interface out of this code? So someone can call the function and pass parameters as input.